build-system/compile/compile-wrappers.js

```diff
@@ -20,7 +20,9 @@ const {VERSION} = require('./internal-version');
 // at least try to unhide the body.
 exports.mainBinary =
   'var global=self;self.AMP=self.AMP||[];' +
-  'try{(function(_){\n<%= contents %>})(AMP._=AMP._||{})}catch(e){' +
+  'try{(function(_){' +
+  'if (Object.prototype.toString.call(self.AMP)==\'[object Object]\') return;' +
```

Comment:
@jridgewell would `Array.isArray(self.AMP)` still be necessary here?

Reply:
It'd actually be smaller: `if (self.AMP && !Array.isArray(self.AMP)) return`

Reply:
done. and removed all spaces as you suggested